src/Nether.Analytics/FileOutputManager.cs

```diff
@@ -72,19 +72,42 @@ private string GetFilePath(string pipelineName, int idx, Message msg)
                     if (File.Exists(filePath))
                     {
                         await AppendMessageToFileAsync(serializedMessage, filePath);
-
                         tryAgain = false;
                     }
                     else
                     {
                         Directory.CreateDirectory(Path.GetDirectoryName(filePath));
 
+                        // append the header to the file
                         string header = $"{_serializer.Header}{Environment.NewLine}";
-
                         await AppendMessageToFileAsync(header, filePath);
                     }
                 }
-                catch (Exception)
+                catch (Exception e)
+                {
+                    // it is possible that another thread is now creating the file and adding the header
+                    // wait a while before continue to try and write the file
+                    await Task.Delay(1000);
+                }
+            } while (tryAgain);
+        }
+
+        private async Task AppendMessageToFileWithoutHeaderAsync(string serializedMessage, string filePath)
+        {
+            var tryAgain = true;
+
+            do
+            {
+                try
+                {
+                    await AppendMessageToFileAsync(serializedMessage, filePath);
+                    tryAgain = false;
+                }
+                catch (DirectoryNotFoundException e)
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+                }
+                catch (Exception e)
                 {
                     // it is possible that another thread is now creating the file and adding the header
```

Comment:
Any comment that isn't up to date, confuses a lot. There is no headers used in this use-case. Explain why we are catching any uncatched exceptions and do something valuable with it, otherwise don't catch exceptions.